cirq/ops/projectors_test.py

```diff
@@ -0,0 +1,166 @@
+import math
+import re
+
+import numpy as np
+import pytest
+
+import cirq
+
+
+def test_projector1d_circuit_diagram():
+    q = cirq.NamedQubit('q')
+    projector = cirq.Projector([[1.0, 0.0]])
+    cirq.testing.assert_has_diagram(cirq.Circuit(projector(q)),
+                                    "q: ───Proj([[1.0, 0.0]])───",
+                                    precision=None)
+    cirq.testing.assert_has_diagram(cirq.Circuit(projector(q)),
+                                    "q: ───Proj([[1.0, 0.0]])───",
+                                    precision=2)
+
+
+def test_projector2d_circuit_diagram():
+    q1 = cirq.NamedQubit('q1')
+    q2 = cirq.NamedQubit('q2')
+    projector = cirq.Projector([[1.0, 0.0, 0.0, 0.0]], qid_shape=(
+        2,
+        2,
+    ))
+    cirq.testing.assert_has_diagram(cirq.Circuit(projector(
+        q1, q2)), ("q1: ───Proj([[1.0, 0.0, 0.0, 0.0]])───\n" + "       │\n" +
```

Comment:
typically we use the multiline string for this, so it's more readable, i.e.
```
"""
q1: ───Proj([[1.0, 0.0, 0.0, 0.0]])───
                 |
q2: ───Proj────────────"""[1:]
```
or similar, check the code there are ton of examples. 

Also, I think we can do better than this, in case the vectors are easy to write in the dirac formalism, then we should write them out that way (using `cirq.dirac_notation`), e.g. this particular example could be something like this - also P would be shorter...though it might be confused with a phase gate?  

```
q1: ───P(|00⟩)──
                  |
q2: ───P──
```

What do you think?

Reply:
Done